src/component.cpp

```diff
@@ -139,10 +139,24 @@ void Component::removeMath()
     mPimpl->mMath.clear();
 }
 
-void Component::addVariable(const VariablePtr &variable)
+bool Component::addVariable(const VariablePtr &variable)
 {
+    // Prevent adding multiple times to list.
+    auto component = shared_from_this();
+    if (component->hasVariable(variable)) {
+        return false;
+    }
```

Comment:
This looks a bit weird to me, I would simply have:
```
if (hasVariable(variable)) {
    return false;
}
```

Reply:
Done in 2b2884ad.